test/e2e/tests/hardware-wallets/ledger/mocks.ts

```diff
@@ -0,0 +1,658 @@
+          suggestedMaxFeePerGas: '30',
+          minWaitTimeEstimate: 15000,
+          maxWaitTimeEstimate: 30000,
+        },
+        estimatedBaseFee: '18',
+        networkCongestion: 0.5,
+        latestPriorityFeeRange: ['1', '2'],
+        historicalPriorityFeeRange: ['1', '3'],
+        historicalBaseFeeRange: ['15', '25'],
+        priorityFeeTrend: 'stable',
+        baseFeeTrend: 'stable',
+      },
+    }));
+}
+
+// Removed complex mockSwapTrades function - using simple mockLedgerEthDaiTrade instead
+
+// Removed mockSmartTransactionAPIs function since Smart Transactions are disabled in the test
+
+async function mockLedgerTransactionRequestsBase(mockServer: MockttpServer) {
+  await mockMultiNetworkBalancePolling(mockServer);
+
+  await mockServerJsonRpc(mockServer, [
+    ['eth_blockNumber'],
+    ['eth_getBlockByNumber'],
+    ['eth_chainId', { result: `0x539` }], // localhost chainId
+    // Mock eth_sendTransaction for regular transaction submission (when Smart Transactions are disabled)
+    [
+      'eth_sendTransaction',
+      {
+        result:
+          '0xe3e223b9725765a7de557effdb2b507ace3534bcff2c1fe3a857e0791e56a518',
+      },
+    ],
+    // Mock eth_getTransactionReceipt for transaction status checking
+    [
+      'eth_getTransactionReceipt',
+      {
+        result: {
+          transactionHash:
+            '0xe3e223b9725765a7de557effdb2b507ace3534bcff2c1fe3a857e0791e56a518',
+          status: '0x1', // This should be '0x1' for success
+          blockNumber: '0x1', // Ensure block number is present
+          blockHash:
+            '0x1111111111111111111111111111111111111111111111111111111111111111',
+          transactionIndex: '0x0',
+          from: '0xF68464152d7289D7eA9a2bEC2E0035c45188223c',
+          to: '0x881d40237659c251811cec9c364ef91dc08d300c',
+          gasUsed: '0x5208',
+          cumulativeGasUsed: '0x5208',
+          contractAddress: null,
+          logs: [],
+        },
+      },
+    ],
+    // Mock eth_getBalance for account balance checking
+    ['eth_getBalance', { result: '0x1158e460913d00000' }], // 20 ETH in wei
+    // Mock eth_gasPrice for gas price fallback
+    ['eth_gasPrice', { result: '0x174876e800' }], // 100 gwei
+    // Mock eth_estimateGas for gas estimation
+    ['eth_estimateGas', { result: '0x5208' }], // 21000 gas
+  ]);
+
+  // Removed Smart Transactions API mocks since Smart Transactions are disabled
+
+  await mockServer
+    .forJsonRpcRequest({
+      method: 'eth_getTransactionReceipt',
+      params: [TRANSACTION_HASH],
+    })
+    .thenJson(200, GET_TRANSACTION_RECEIPT_RESPONSE);
+
+  await mockServer
+    .forJsonRpcRequest({
+      method: 'eth_getTransactionByHash',
+      params: [TRANSACTION_HASH],
+    })
+    .thenJson(200, GET_TRANSACTION_BY_HASH_RESPONSE);
+
+  await mockServer
+    .forJsonRpcRequest({
+      method: 'eth_getBlockByHash',
+      params: [BLOCK_HASH],
+    })
+    .thenJson(200, GET_BLOCK_BY_HASH_RESPONSE);
+}
```

Comment:
this is not needed -- we are using anvil localhost, therefore mocks don't work, we simply need to rely on the real local network RPC calls
```suggestion
```